### DISCUSSION TITLE
IconLayer data contains multiple duplicate coordinates

### DISCUSSION THREAD

**Discussion by kamrankhanwastaken** [visgl/deck.gl#5887] [Q&A]
Hey everyone,

I was wondering if there is a simple solution to a visualization/user experience issue I'm having.

I'm using the IconLayer to show data that can sometimes overlap i.e. two or more items in the data for IconLayer could be at the exact same coordinates. This unfortunately doesn't actually work well visually as there is no way to tell that a particular coordinate has more than one pin.

Does anyone have any suggestions on how this visualization can be improved? I'm wondering if there is an easy way to cluster multiple pins to indicate that more than one pin is present but I'm a little lost if this is something I can do with IconLayer's native capabilities or if I will need to do this on my own.

![Screen-Shot-2021-06-16-at-12 43 25-PM](https://user-images.githubusercontent.com/9968974/122282836-b879a080-cea0-11eb-93bc-dae7582b2a66.png)

I'm using deckgl with MapBox in React to render layers.

**Comment by Pessimistress** [comment]
https://deck.gl/examples/icon-layer/
